sale_partner_address_restrict/models/sale_order.py

```diff
@@ -0,0 +1,39 @@
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.constrains("partner_id", "partner_invoice_id", "partner_shipping_id")
+    def _check_partner_addresses(self):
+        for order in self:
+            if order.partner_id and (
```

Comment:
IMHO, as this changes the standard behavior, a configuration parameter should be set to activate the restriction.

Reply:
But you would not install the module if you don't want to have the restriction, don't you think?

Reply:
That has always be a debate :smile: 

But you maybe want to decouple deployment flows and activation. Not changing standard behavior by installing a module is always a good practice.

This allows also better flexibility.

And for our tests flows, that allows to avoid separate job flows for conflicting modules.